docs/faq/DesignDecisions.md

```diff
@@ -0,0 +1,75 @@
+
+A potential use-case for using the action inside a subscriber -- which is an unsupported feature -- is to ensure that a component only re-renders after certain kinds of actions. Re-rendering should instead be controlled instead through:
+1.) the [shouldComponentUpdate](https://facebook.github.io/react/docs/react-component.html#shouldcomponentupdate) lifecycle method
+2.) the [virtual DOM equality check (vDOMEq)](https://facebook.github.io/react/docs/optimizing-performance.html#avoid-reconciliation)
+3.) [React.PureComponent](https://facebook.github.io/react/docs/optimizing-performance.html#examples)
+4.) Using React-Redux: use [mapStateToProps](https://github.com/reactjs/react-redux/blob/master/docs/api.md#connectmapstatetoprops-mapdispatchtoprops-mergeprops-options) to subscribe components to only the parts of the store that they need.
+
+<a id="does-not-support-classes"></a>
+### Why doesn't Redux support using classes for actions and reducers?
+The pattern of using functions, called action creators, to return action objects may seem counterintuitive to programmers with a lot of Object Oriented Programming experience, who would see this is a strong use-case for Classes and instances. Class instances for action objects and reducers are not supported because class instances make serialization and deserialization tricky. Deserialization methods like JSON.parse(string) will return a plain old Javascript object rather than class instances. 
+
+Serialization enables the brower to store all actions that have been dispatched, as well as the previous store states, with much less memory. Rewinding and 'hot reloading' the store is central to the Redux developer experience and the function of Redux DevTools. This also enables deserialized actions to be stored on the server and re-serialized in the brower in the case of server-side rendering with Redux.
+
+<a id="why-currying"></a>
+### Why does the middleware signature use currying?
+The [curried function signature](https://github.com/reactjs/redux/issues/1744) of declaring middleware is [deemed unnecessary](https://github.com/reactjs/redux/pull/784) by some, because both store and next are available when the applyMiddleware function is executed. This issue has been determined to not be [worth introducing breaking changes](https://github.com/reactjs/redux/issues/1744).
+
+<a id="closure-dispatch"></a>
+### Why does applyMiddleware use a closure for dispatch?
+applyMiddleware takes the existing dispatch from the store and closes over it to create the initial chain of middlewares that have been invoked with an object that exposes the getState and dispatch functions, which enables middlewares that [rely on dispatch during initialization](https://github.com/reactjs/redux/pull/1592) to run. 
+
+<a id="combineReducers-limitations"></a>
+### Can you please change combineReducers to support nested state trees?
```

Comment:
Let's change the title on this one.  Maybe something like: "Why doesn't `combineReducers` include a third argument with the entire state?"

I'd also like to see the answer fleshed out a bit more.  Overall, the general answer is that `combineReducers` is deliberately opinionated, and it's also not immediately obvious what a potential third arg should be (entire state tree, some callback, some other part of the state tree, ... ?).  We _might_ change that in the future, but for now the answer is "if it doesn't fit your use case, you can write your own".